tools/cert-sidecar/index.js

```diff
@@ -0,0 +1,83 @@
+const {
+  ConfigManager,
+  Logger,
+} = require('@dojot/microservice-sdk');
+
+const util = require('util');
+
+
+// Creating the configuration
+const userConfigFile = process.env.CERT_SC_USER_CONFIG_FILE || 'production.conf';
+ConfigManager.loadSettings('CERT_SC', userConfigFile);
+const config = ConfigManager.getConfig('CERT_SC');
+
+// Logger configuration
+const {
+  log: configLog, app: configApp,
+} = config;
+Logger.setTransport('console', {
+  level: configLog['console.level'],
+});
+Logger.setVerbose(configLog['console.verbose']);
+if (configLog.file) {
+  Logger.setTransport('file', {
+    level: configLog['file.level'],
+    filename: configLog['file.filename'],
+  });
+}
+const logger = new Logger(`cert-sc-${configApp['sidecar.to']}:index`);
+
+logger.info(`The final configuration:\n${util.inspect(config, false, 5, true)}`);
+
+const App = require('./app/App');
+const ServiceStateMgmt = require('./app/ServiceStateMgmt');
+
+const unhandledRejections = new Map();
```

Comment:
This strategy of keeping unhandled rejections in a map might be problematic, because the application might be running in an unknown state, and it the unhandled rejection might be never handled in future loops. 

My suggestion is to keep the things simpler as possible. Catch the 'unhandledRejection' and call process.kill().

A more elaborated solution is to monitor the map and if a promise is kept there for a certain time, ie, it has not be handled, call process.kill().